src/PollingBlockTracker.ts

```diff
@@ -303,25 +279,53 @@ export class PollingBlockTracker
     }
     return res.result;
   }
-}
 
-/**
- * Waits for the specified amount of time.
- *
- * @param duration - The amount of time in milliseconds.
- * @param unref - Assuming this function is run in a Node context, governs
- * whether Node should wait before the `setTimeout` has completed before ending
- * the process (true for no, false for yes). Defaults to false.
- * @returns A promise that can be used to wait.
- */
-async function timeout(duration: number, unref: boolean) {
-  return new Promise((resolve) => {
-    const timeoutRef = setTimeout(resolve, duration);
-    // don't keep process open
-    if (timeoutRef.unref && unref) {
+  private async _updateAndQueue() {
+    let interval = this._pollingInterval;
+
+    try {
+      await this._updateLatestBlock();
+    } catch (err: any) {
+      const newErr = new Error(
+        `PollingBlockTracker - encountered an error while attempting to update latest block:\n${
+          err.stack ?? err
+        }`,
+      );
+
+      try {
+        this.emit('error', newErr);
+      } catch (emitErr) {
```

Comment:
Why do we use `try/catch` at this call site `this.emit` but not for others in this same file? 

Related question: should we move this `try/catch` into the implementation of `emit` so that clients can safely call `emit` without needing to `try/catch`?

Reply:
My assumption is we have to be more cautious of errors here since this is our core polling function so we don't want it to fail and the loop to stop.

I'd feel more comfortable with any change to this logic to be in a separate PR to limit the scope of this one to the fix.